python/llm/src/bigdl/llm/transformers/low_bit_linear.py

```diff
@@ -440,7 +440,8 @@ def forward(self, x: torch.Tensor):
                 # current workaround to reduce first token latency of fp32 input
                 # sometimes fp16 cause nan and training instability
                 # disable the conversion when training
-                if self.conver_to_half and x_2d.shape[0] > 1 and x_2d.dtype == torch.float32:
+                if self.conver_to_half and x_2d.shape[0] > 1 and x_2d.dtype == torch.float32 and \
+                        (x_2d.shape[0] > 8 or self.qtype not in [SYM_INT4, SYM_INT8, FP8]):
```

Comment:
Add a function (e.g., `using_xmx`) to check the conditions (e.g., GPU type, batch size, data types, etc.)

Reply:
done